File: libraries/apollo-runtime-java/src/main/java/com/apollographql/apollo3/runtime/java/interceptor/internal/AutoPersistedQueryInfo.java
package com.apollographql.apollo.runtime.java.interceptor.internal;

import com.apollographql.apollo.api.ExecutionContext;
import org.jetbrains.annotations.NotNull;

class AutoPersistedQueryInfo implements ExecutionContext.Element {
  public boolean hit;

  public AutoPersistedQueryInfo(boolean hit) {
    this.hit = hit;
  }

  @NotNull @Override public Key<?> getKey() {
    return KEY;
  }

  static final ExecutionContext.Key<AutoPersistedQueryInfo> KEY = new ExecutionContext.Key<AutoPersistedQueryInfo>() {};
}

File: libraries/apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo/AutoPersistedQueryInfo.kt
package com.apollographql.apollo

import com.apollographql.apollo.api.ApolloResponse
import com.apollographql.apollo.api.ExecutionContext
import com.apollographql.apollo.api.Operation

/**
 * Information about auto persisted queries
 *
 * @param hit: whether the query was hit or required another round trip to send the document
 */
class AutoPersistedQueryInfo(
    val hit: Boolean,
) : ExecutionContext.Element {
  override val key: ExecutionContext.Key<*>
    get() = Key

  companion object Key : ExecutionContext.Key<AutoPersistedQueryInfo>
}

val <D : Operation.Data> ApolloResponse<D>.autoPersistedQueryInfo
  get() = executionContext[AutoPersistedQueryInfo]

